### PR TITLE
ws_eink: Add revert 8bits in byte and remove unnecessary ssbuf

### PR DESCRIPTION
**Pull request by nxthongbk**
Add revert 8bits in byte to fix issue mirror with littlevgl and
remove unnecessary ssbuf. Using vmem only to get data from mmap.

### CODE REVIEW

**Comment by dpfrey** [comment]
In your previous pull request (#21), I re-based and squashed the last 2 commits from your `update-eink` branch and put them onto the mangOH master branch.  This means that I don't need (or want) the two unsquashed versions of those commits to appear in the history of the mangOH repository.

gitk

As you can see from the image, the commits that I squashed as well as a no-op merge commit appear in the graph.  I would prefer that you create a new branch for each pull request.

File: experimental/waveshare_eink/linux_kernel_modules/fb_waveshare_eink.c [hunk 1]
```diff
@@ -72,6 +72,16 @@ const u8 lut_partial_update[] = {
 	0x00, 0x00, 0x00, 0x00, 0x00, 0x00
 };
 
+static u8 bitrev8(u8 byte)
```
Comment (dpfrey) [inline experimental/waveshare_eink/linux_kernel_modules/fb_waveshare_eink.c]: `bitrev8` is a standard linux function.  See [`here`](https://elixir.bootlin.com/linux/v3.18.136/source/include/linux/bitrev.h#L8).  Please `#include <linux/bitrev.h>` and use that version.

File: experimental/waveshare_eink/linux_kernel_modules/fb_waveshare_eink.c [hunk 2]
```diff
@@ -339,12 +349,13 @@ static int ws_eink_init_display(struct ws_eink_fb_par *par)
...
 	int ret = 0;
+	int i;
+	int frame_size;
 	u8 *vmem = par->info->screen_base;
-	u8 *ssbuf = par->ssbuf;
 	const u8 *lut;
 	size_t lut_size;
 	static int update_count = 0;
-
+
```
Comment (dpfrey) [inline experimental/waveshare_eink/linux_kernel_modules/fb_waveshare_eink.c]: Trailing whitespace

File: experimental/waveshare_eink/linux_kernel_modules/fb_waveshare_eink.c [hunk 3]
```diff
@@ -358,9 +369,12 @@ static int ws_eink_update_display(struct ws_eink_fb_par *par)
 	if (ret)
 		return ret;
 
-	memcpy(&ssbuf, &vmem, sizeof(vmem));
+	frame_size = par->props->height * par->props->width * par->props->bpp / 8;
 
-	ret = set_frame_memory(par, ssbuf);
+	for (i = 0; i < frame_size; i++) {
```
Comment (dpfrey) [inline experimental/waveshare_eink/linux_kernel_modules/fb_waveshare_eink.c]: I think this won't work in the general case.  There is no guarantee that the entire `vmem` has been rewritten each time `ws_eink_update_display` is called.  I think what you need to do is create a local copy of `vmem` and call `bitrev8` on each byte of that copy.  Then pass that copy to `set_frame_memory`.  Once `set_frame_memory` is done, the temporary bit reversed copy of `vmem` can be discarded.

**Comment by nxthongbk** [comment]
Hi David

I close this pull request because of new pull request is created. 
New branch is added.

Thanks
Thong Nguyen

**Pull request closed without merging**